packages/gestalt/src/Masonry/defaultTwoColumnModuleLayout.js

```diff
@@ -323,34 +360,50 @@ const defaultTwoColumnModuleLayout = <T: { +[string]: mixed }>({
         (item) => !item.columnSpan || item.columnSpan === 1,
       );
 
-      const isFirstRow = heights.some((height) => height === 0);
-      const multiColumnItemWidth = parseInt(twoColumnItems[0].columnSpan, 10);
-      const availableSlotsOnFirstRow = heights.reduce(
-        (acc, height) => (height === 0 ? acc + 1 : acc),
-        0,
-      );
-      const fitsFirstRow = availableSlotsOnFirstRow >= multiColumnItemWidth + twoColumnIndex;
+      // The empty columns can be different from columnCount if there are
+      // items already positioned from previous batches
+      const emptyColumns = heights.reduce((acc, height) => (height === 0 ? acc + 1 : acc), 0);
+
+      const multiColumnItemColumnSpan = parseInt(twoColumnItems[0].columnSpan, 10);
+      const fitsFirstRow = emptyColumns >= multiColumnItemColumnSpan + twoColumnIndex;
```

Comment:
this makes more sense thank you - i honestly think if we just applied ^ changes to the last commit, that would be enough/more clear already